### ISSUE TITLE
Make a default set of options

### ISSUE THREAD

**Issue by odow** [cvanaret/Uno#28]
Rather than force everyone to schlep around the full uno.options, it makes more sense to define the current `/uno.options` file at the default, and then if people provide a partial uno.options file it modifies only the options they provide.

We should then also change the printing to print only non-default options.

I don't want to see
```
julia> Uno_jll.amplexe() do exe
           run(`$exe problem.nl`)
       end
Problem problem.nl_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
- LS_backtracking_ratio = 0.5
- LS_min_step_length = 1e-12
- LS_scale_duals_with_step_length = yes
- armijo_decrease_fraction = 1e-4
- armijo_tolerance = 1e-9
- barrier_damping_factor = 1e-5
- barrier_default_multiplier = 1
- barrier_initial_parameter = 0.1
- barrier_k_epsilon = 10
- barrier_k_mu = 0.2
- barrier_k_sigma = 1e10
- barrier_push_variable_to_interior_k1 = 1e-2
- barrier_push_variable_to_interior_k2 = 1e-2
- barrier_regularization_exponent = 0.25
- barrier_small_direction_factor = 10.
- barrier_tau_min = 0.99
- barrier_theta_mu = 1.5
- barrier_update_fraction = 10
- constraint_relaxation_strategy = feasibility_restoration
- dual_regularization_fraction = 1e-8
- filter_beta = 0.999
- filter_capacity = 50
- filter_fact = 1.25
- filter_gamma = 0.001
- filter_type = standard
- filter_ubd = 1e2
- globalization_mechanism = LS
- globalization_strategy = fletcher_filter_method
- l1_constraint_violation_coefficient = 1
- least_square_multiplier_max_norm = 1e3
- linear_solver = MUMPS
- logger = INFO
- loose_tolerance = 1e-6
- loose_tolerance_consecutive_iteration_threshold = 15
- max_iterations = 2000
- primal_regularization_decrease_factor = 3.
- primal_regularization_fast_increase_factor = 100.
- primal_regularization_initial_factor = 1e-4
- primal_regularization_lb = 1e-20
- primal_regularization_slow_increase_factor = 8.
- progress_norm = L1
- protect_actual_reduction_against_roundoff = no
- regularization_failure_threshold = 1e40
- residual_norm = INF
- residual_scaling_threshold = 100.
- scale_functions = no
- sparse_format = COO
- statistics_LS_step_length_column_order = 10
- statistics_barrier_parameter_column_order = 8
- statistics_complementarity_column_order = 105
- statistics_major_column_order = 1
- statistics_minor_column_order = 2
- statistics_objective_column_order = 100
- statistics_print_header_every_iterations = 15
- statistics_regularization_column_order = 21
- statistics_restoration_phase_column_order = 20
- statistics_stationarity_column_order = 104
- statistics_status_column_order = 200
- statistics_step_norm_column_order = 31
- subproblem = primal_dual_interior_point
- switch_to_optimality_requires_linearized_feasibility = yes
- switching_delta = 0.999
- switching_infeasibility_exponent = 2
- threshold_unsuccessful_attempts = 8
- time_limit = inf
- tolerance = 1e-8
- unbounded_objective_threshold = -1e20
```

I want to see
```
julia> Uno_jll.amplexe() do exe
           run(`$exe problem.nl`)
       end
Problem problem.nl_equalityconstrained_boundrelaxed
1 variables, 0 constraints

Options:
- globalization_mechanism = LS
- linear_solver = MUMPS
- subproblem = primal_dual_interior_point
```

**Comment by cvanaret** [comment]
This is a fair point. I can definitely hard code the default options and they will be overwritten if an option file is provided.
By default, Uno prints all the options that are used, which should help with reproducibility.

**Comment by cvanaret** [comment]
Implemented in commit 40420f0.
The default options are hard-coded in `Options.cpp` and can be overwritten via the command line such as:
```shell
./uno_ampl model.nl -AMPL preset=filtersqp option_file=partial_option_file subproblem=LP
```
